adapter-tcp/src/main/java/io/nosqlbench/adapter/tcpserver/TcpServerAdapterSpace.java

```diff
@@ -0,0 +1,291 @@
+                        "as a pattern (regex) to subselect a set of bindings by name. You can simply use the name of a binding\n" +
+                        "here as well.")
+
+            )
+            .asReadOnly();
+    }
+    private interface Shutdown {
+        void shutdown();
+    }
+    public static class SocketWriter implements Runnable, Shutdown {
+        private final BlockingQueue<String> sourceQueue;
+        private final OutputStream outputStream;
+        private final OutputStreamWriter writer;
+        private boolean running = true;
+
+
+        public SocketWriter(BlockingQueue<String> sourceQueue, Socket connectedSocket) {
+            this.sourceQueue = sourceQueue;
+            try {
+                outputStream = connectedSocket.getOutputStream();
+                this.writer = new OutputStreamWriter(outputStream);
+                //connectedSocket.shutdownInput();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        public void shutdown() {
+            this.running = false;
+        }
+
+        @Override
+        public void run() {
+            try (Writer writer = this.writer) {
+                while (running) {
+                    while (!sourceQueue.isEmpty() ) {
+                        try {
+                            String data = sourceQueue.take();
+                            writer.write(data);
+                            writer.flush();
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    try {
+                        Thread.sleep(10);
+                    } catch (InterruptedException ignored) {
+                    }
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+            finally {
+                //TODO: Add the shutdown logic to do after closing the socket
+            }
+
+        }
+
+    }
+
+    public static class QueueWriterAdapter extends Writer {
+        private BlockingQueue<String> queue;
+
+        public QueueWriterAdapter(BlockingQueue<String> queue) {
+            this.queue = queue;
+        }
+
+        @Override
+        public synchronized void write( char[] cbuf, int off, int len) {
+            while (true) {
+                try {
+                    queue.put(new String(cbuf, off, len));
+                    return;
+                } catch (InterruptedException ignored) {
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        @Override
+        public synchronized void flush() throws IOException {
+        }
+
+        @Override
+        public synchronized void close() throws IOException {
+            flush();
```

Comment:
Let's put a `super.flush()` so the base class logic get invoked.
This can thrown an exception so it might be good to catch this and log as an error, and then set the queue to null.

Reply:
This is a good idea, but then when I tried to implement it I discovered that flush() is only an abstract in the base class (java Writer). so it doesn't really do any extra logic in the base class, this also means that our call to "flush()" in close isn't really doing anything right now.